icu4c/source/common/uloc.cpp

```diff
@@ -566,6 +566,7 @@ static int32_t locale_canonKeywordName(char *buf, const char *keywordName, UErro
   int32_t keywordNameLen = 0;
 
   for (; *keywordName != 0; keywordName++) {
+    // TODO: fix when the keyword = "measurement-unit".
     if (!UPRV_ISALPHANUM(*keywordName)) {
```

Comment:
@sffc , @markusicu 

Because the keywords can only contains numbers or letters, I could not use "measurement-unit" as a key in `Locale::getKeywordValue`. even that "measurement-unit" is the key to get the desired value.

Do you have a solution?

Reply:
Sure, I have removed it.